blog post for knative release 0.22 (#3486)
knative/docs: blog/releases/announcing-knative-v0-22-release.md

---
title: "Version v0.22 release"

> linkTitle: "Version v0.22 release"

Reviewer: Do we need "Version" and "v" in the title?

Author: I will remove the small `v`

Author: "[Carlos Santana](https://twitter.com/csantanapr)"
Author handle: https://github.com/csantanapr
date: 2021-04-06
description: "Knative v0.22 release announcement"
type: "blog"
image: knative-eventing.png
---

### Announcing Knative v0.22 Release

A new version of Knative is now available across multiple components.

> Follow the instructions in the documentation [Installing Knative](https://knative.dev/docs/install/) for the respective component.

Reviewer 2: ```suggestion
To upgrade, follow the steps in [Upgrading Your Installation](https://knative.dev/docs/install/upgrade-installation/) for the relevant components.
```

#### Table of Contents
- [Serving v0.22](#serving-v022)
- [Eventing v0.22](#eventing-v022)
- Eventing Extensions
    - [Eventing Kafka Broker v0.22](#eventing-kafka-broker-v022)
- `kn` [CLI v0.22](#client-v022)
    - `kn` [CLI Plugins](#cli-plugins)
- [Operator v0.22](#operator-v022)
- [Thank you contributors](#thank-you-contributors)

### Highlights

> - Serving Domain Mapping improves multi-tenency support to avoid domain name 'squatting' in cluster.

Reviewer 3: ```suggestion
- Serving Domain Mapping improves multi-tenancy support to avoid domain name 'squatting' in the cluster.
```

> - Serving Domain Mapping improves multi-tenency support to avoid domain name 'squatting' in cluster.

Reviewer: Don't use 'squatting' here; explain properly what it means

> - Eventing now allows Trigger users to set another namespace's Subscriber

Reviewer: ```suggestion
- Eventing now allows subscribers and triggers from different namespaces to be used together.
```

Not totally sure if this is right or what this means otherwise, it seems like you're saying triggers can have subscribers that are in a different namespace to them now?

Author: yes, I think so

> - Eventing kafka broker now Kubernetes's minimum version is 1.18

Reviewer 3: I don't understand this sentence. Maybe because of my lack of understanding of Eventing.

> - Eventing kafka broker now Kubernetes's minimum version is 1.18

Reviewer 3: ```suggestion
- Eventing Kafka broker now Kubernetes's minimum version is 1.18
```

> - Eventing kafka broker now Kubernetes's minimum version is 1.18

Reviewer: ```suggestion
- 1.18 is now the minimum Kubernetes version required to use the Apache Kafka broker with Knative Eventing v0.22.
```

> - Eventing kafka now supports the ability to choose between ordered and unordered delivery

Reviewer 3: ```suggestion
- Eventing Kafka now supports the ability to choose between ordered and unordered delivery
```

Reviewer: Use Apache Kafka everywhere, or just "Kafka with Knative Eventing"?

Reviewer: Also, be consistent about using periods / full stops at the end of sentences 😉

- The CLI `kn` 0.22.0 comes with some bug fixes and minor feature enhancements. It's mostly a polishing release. If you are using the client API, there is a breaking change that was needed to align with Kubernetes client API's

> - There are two new CLI plugins align with 0.22 release, [kn-plugin-admin](https://github.com/knative-sandbox/kn-plugin-admin) and [kn-plugin-source-kafka](https://github.com/knative-sandbox/kn-plugin-source-kafka)

Reviewer 3: ```suggestion
- There are two new CLI plugins to align with the 0.22 release, [kn-plugin-admin](https://github.com/knative-sandbox/kn-plugin-admin) and [kn-plugin-source-kafka](https://github.com/knative-sandbox/kn-plugin-source-kafka)
```

> - The CLI `kn` 0.22.0 comes with some bug fixes and minor feature enhancements. It's mostly a polishing release. If you are using the client API, there is a breaking change that was needed to align with Kubernetes client API's
> - There are two new CLI plugins align with 0.22 release, [kn-plugin-admin](https://github.com/knative-sandbox/kn-plugin-admin) and [kn-plugin-source-kafka](https://github.com/knative-sandbox/kn-plugin-source-kafka)

Reviewer: ```suggestion
- Version 0.22 of the `kn` CLI contains bug fixes and minor feature enhancements.
- A breaking change was required to align the client API with the Kubernetes client API.
- There are two new CLI plugins available in version 0.22; [kn-plugin-admin](https://github.com/knative-sandbox/kn-plugin-admin) and [kn-plugin-source-kafka](https://github.com/knative-sandbox/kn-plugin-source-kafka).
```

Reviewer: @omerbensaadon @snneji @RichardJJG can we choose one way to consistently refer to "version" across docs and Knative in general? Do you have any preference? I don't want to keep using "version 0.22" in some places and "v0.22" in others for example if we can help it. Can you maybe take this to the TOC, Omer?

Author: My opinion is that deciding to use 0.22 vs. v0.22 in the website/docs, is something that the docs/ux team can make the decision and document that this is the style across the website.

I added the question in the new doc that @omerbensaadon started https://docs.google.com/document/d/1tigolRzKhronvgoAJfszE-XHWWd0LQ8o_j0D73O9KvM/edit

Author: I'm making everything consistent for now to use `v0.22`

> - The Knative Operator release with some bug fixes, and supports the latest version of knative serving and eventing.

Reviewer 3: ```suggestion
- The Knative Operator was released with some bug fixes and supports the latest version of Knative Serving and Eventing.
```

> - The Knative Operator release with some bug fixes, and supports the latest version of knative serving and eventing.

Reviewer: ```suggestion
- The Knative Operator 0.22 release contains bug fixes, and supports version 0.22 of Knative Serving and Knative Eventing.
```

### Serving v0.22

#### 💫 New Features & Changes

- Added an autoscaling annotation to choose a different aggregation algorithm for the autoscaling metrics. This is experimental currently. [#10840](https://github.com/knative/serving/pull/10840)

> - Added autocreateClusterDomainClaims flag to network config map. [networking/#330](https://github.com/knative/networking/pull/330)

Reviewer: ```suggestion
- Added the `autocreateClusterDomainClaims` flag to the network ConfigMap. [networking/#330](https://github.com/knative/networking/pull/330)
```

#### 🐞 Bug Fixes

> - Adds validation that a default max-scale is set if a max-scale-limit is specified in the autoscaler configmap (since otherwise the default max-scale, i.e. 0 = no max, would fail validation as it is above the max-scale-limit). [#10921](https://github.com/knative/serving/pull/10921)

Reviewer: ```suggestion
- Added validation that a default `max-scale` is set if a `max-scale-limit` is specified in the autoscaler ConfigMap. Otherwise, the default `max-scale`, `0` = no max, would fail validation as it is above the `max-scale-limit`. [#10921](https://github.com/knative/serving/pull/10921)
```

> - Bumped the resource request and limits of the autoscaler to 100m/100Mi, 1000m/1000Mi respectively. [#10865](https://github.com/knative/serving/pull/10865)

Reviewer: ```suggestion
- Raised the resource request and limits of the autoscaler to 100m/100Mi, 1000m/1000Mi respectively. [#10865](https://github.com/knative/serving/pull/10865)
```

> - Fixed a regression where the pod bringup time might have a latency of 10s or more even though the container should be up quickly. [#10992](https://github.com/knative/serving/pull/10992)

Reviewer: ```suggestion
- Fixed a regression where the latency of starting a pod could be 10s or more. [#10992](https://github.com/knative/serving/pull/10992)
```

> - Reduced the necessary memory allocations in the activator significantly, especially with disabled tracing. [#11016](https://github.com/knative/serving/pull/11016), [#11013](https://github.com/knative/serving/pull/11013), [#11009](https://github.com/knative/serving/pull/11009), [#11008](https://github.com/knative/serving/pull/11008)

Reviewer: ```suggestion
- Reduced required memory allocations in the activator significantly, particularly when tracing is turned off. [#11016](https://github.com/knative/serving/pull/11016), [#11013](https://github.com/knative/serving/pull/11013), [#11009](https://github.com/knative/serving/pull/11009), [#11008](https://github.com/knative/serving/pull/11008)
```

> - Fix the incorrect Gateway name format for DomainMapping auto TLS feature for net-istio implmenetation. [net-istio#532](https://github.com/knative-sandbox/net-istio/pull/532)

Reviewer: ```suggestion
- Fixed the incorrect gateway name format for the domain mapping auto TLS feature for `net-istio` implementation. [net-istio#532](https://github.com/knative-sandbox/net-istio/pull/532)
```

### Eventing v0.22

#### 🚨 API Changes

> - v1alpha1 Channel duck types are no longer supported. [#5005](https://github.com/knative/eventing/pull/5005)

Reviewer: ```suggestion
- `v1alpha1` channel duck types are no longer supported. [#5005](https://github.com/knative/eventing/pull/5005)
```

> - Allow Trigger users to set another namespace's Subscriber, And when Subscriber's namespace not been set, it will be set to Trigger's namespace [#4995](https://github.com/knative/eventing/pull/4995)

Reviewer: Again, it's not clear to me what you're trying to say here re "set another namespace's Subscriber".

From a docs perspective;
- Please use lowercase for triggers, subscribers
- Do no use possessive ('s) for namespace and other inanimate objects. E.g. "If the namespace for a subscriber has not been set, the subscriber defaults to using the namespace of the trigger."

Author: I really don't know what is this trying to say either is not clear to me.
I copy from the release notes here 
https://github.com/knative/eventing/releases/tag/v0.22.1

![image](https://user-images.githubusercontent.com/1094878/115813326-f9819700-a3c0-11eb-8327-8d607e7f86b5.png)

I think what it means is that now Triggers allow specifying a Subscriber in a different namespace, when the 
namespace of the Subscriber is not specified it will default to the same namespace as the Trigger.

@n3wscott is my description accurately above ^^ ?

I will comment it out for now, but I didn't see the docs updated to cover this new enhancement

Reviewer 4: Yes that's correct. Now in the trigger subscriber you can specify the namespace of the subscriber, if it's not specified then it'll default to the same namespace of the trigger.

Reviewer: @devguyio thanks for clarifying! If there's no docs issue to add this change could you open one please?

#### 🐞 Bug Fixes

> - Fix PingSource adapter not always releasing the leader election lease when terminating [#5162](https://github.com/knative/eventing/pull/5162)
> - Fix PingSource sending duplicate events when the leading adapter fails to renew its lease [#4908](https://github.com/knative/eventing/pull/4908)
> - Fix bug preventing the namespace-scoped in-memory channel to become ready. [#4906](https://github.com/knative/eventing/pull/)
> - Fix excessive number of Replicas for eventing-webhook [#5112](https://github.com/knative/eventing/pull/5112)
> - Webhook timeout for API server calls has been increased to 10 seconds. [#5175](https://github.com/knative/eventing/pull/5175)

Reviewer: ```suggestion
- Fixed a bug that caused the PingSource adapter to not always release the leader election lease when terminating. [#5162](https://github.com/knative/eventing/pull/5162)
- Fixed a bug that caused PingSource objects to send duplicate events when the leading adapter failed to renew its lease. [#4908](https://github.com/knative/eventing/pull/4908)
- Fixed a  bug that prevented namespace-scoped InMemoryChannel objects from becoming `READY`. [#4906](https://github.com/knative/eventing/pull/)
- Fixed excessive number of replicas that were being created for `eventing-webhook`. [#5112](https://github.com/knative/eventing/pull/5112)
- The webhook timeout for API server calls has been increased to 10 seconds. [#5175](https://github.com/knative/eventing/pull/5175)
```

#### 🧹 Clean up

> - Do not set the finalizer for pingSource for pingSource controller, just deal with it in reconcilekind [#5002](https://github.com/knative/eventing/pull/5002)

Reviewer: Be consistent with PingSource formatting.

@omerbensaadon this is part of the API object formatting issue for the style guide; we need to make a decision on that ASAP so that folks have some guidance.

> - Do not set the finalizer for pingSource for pingSource controller, just deal with it in reconcilekind [#5002](https://github.com/knative/eventing/pull/5002)

Reviewer: ```suggestion
- Do not set a finalizer for PingSource controllers. Use `reconcilekind` instead. [#5002](https://github.com/knative/eventing/pull/5002)
```

> - Do not set the finalizer for pingSource for pingSource controller, just deal with it in reconcilekind [#5002](https://github.com/knative/eventing/pull/5002)

Reviewer 5: If you make the decision on this, I will document it :D

> - Do not set the finalizer for pingSource for pingSource controller, just deal with it in reconcilekind [#5002](https://github.com/knative/eventing/pull/5002)

Reviewer 5: CC: @RichardJJG @snneji who are also qualified to make a decision on this

### Eventing Extensions

> #### Eventing Kafka Broker v0.22

Reviewer: Decide on consistent name here, @matzew any thoughts? Should this be Apache Kafka broker instead?

Author: I will go calling the broker "Apache Kafka Broker" and follow consistency with what we already have on the website https://knative.dev/docs/eventing/broker/kafka-broker/

#### 🚨 Breaking or Notable

> - Kubernetes's minimum version is 1.18. [#779](https://github.com/knative-sandbox/eventing-kafka-broker/pull/779)

Reviewer: I'd move this to way earlier in the doc; also we call this out separately earlier in the doc for the Kafka broker, so I'm not sure we need to have it in two places? The headings seem a bit messy in this blog in general, there isn't a very logical flow IMO.

Author: I remove it from here and leave it in the earlier "highlights" section

#### 💫 New Features & Changes

- Support ordered delivery.
  - You can choose between ordered and unordered delivery through the label kafka.eventing.knative.dev/delivery.order in your triggers.

>   - Check out the docs for more details at https://knative.dev/docs/eventing/broker/kafka-broker . [#589](https://github.com/knative-sandbox/eventing-kafka-broker/pull/)

Reviewer: Again, I feel like this was mentioned earlier but in less detail. We need a better overall structure for this doc, but we can follow that up in a future release. Won't block the PR over it.

> - Support ordered delivery.
>   - You can choose between ordered and unordered delivery through the label kafka.eventing.knative.dev/delivery.order in your triggers.
>   - Check out the docs for more details at https://knative.dev/docs/eventing/broker/kafka-broker . [#589](https://github.com/knative-sandbox/eventing-kafka-broker/pull/)

Reviewer: ```suggestion
- The Kafka broker now supports ordered delivery. You can choose between ordered and unordered delivery by using the `kafka.eventing.knative.dev/delivery.order` label in the trigger spec. See the [Kafka broker](https://knative.dev/docs/eventing/broker/kafka-broker) documentation. [#589](https://github.com/knative-sandbox/eventing-kafka-broker/pull/)
```

#### 🐞 Bug Fixes

> - Add producer interceptor io.cloudevents.kafka.PartitionKeyExtensionInterceptor to provide ordered delivery based on the partitioning extension of the CloudEvents spec. [#751](https://github.com/knative-sandbox/eventing-kafka-broker/pull/751)
> - Fix unable to deploy KafkaSink without Kafka Broker installed [#714](https://github.com/knative-sandbox/eventing-kafka-broker/pull/714)

Reviewer: ```suggestion
- Added a producer interceptor, `io.cloudevents.kafka.PartitionKeyExtensionInterceptor`, to provide ordered delivery based on the partitioning extension of the CloudEvents spec. [#751](https://github.com/knative-sandbox/eventing-kafka-broker/pull/751)
- Fix unable to deploy KafkaSink without Kafka Broker installed [#714](https://github.com/knative-sandbox/eventing-kafka-broker/pull/714)
```

### Client v0.22

> kn 0.22.0 comes with some bug fixes and minor feature enhancements. It's mostly a polishing release. If you are using the client API, there is a breaking change that was needed to align with Kubernetes client API's

Reviewer: This is mentioned already higher up in the doc, delete entirely.

Author: remove left it in the highlights above

#### META

> The compile dependencies have been updated to Knative Serving 0.22.0 and Knative Eventing 0.22.0. There are no changes in the used API version for communicating with the backend.

Reviewer: Move this to earlier in the doc, also the META heading is not clear. I don't think this needs a whole heading, could probably just go in the highlights section.

Author: I agree I removed it from the doc, is more internal info

#### Managing custom domain mappings

> This release adds support for CRUD management of domain mappings. I.e. you can use `kn domain` along with its sub-commands `create`, `update`, `describe`, `list` and `delete` to fully manage DomainMapping resources for the installation of custom domains for Knative services.
> 

Reviewer: ```suggestion
This release adds support for CRUD management of domain mappings. You can use the `kn domain` CLI command, along with its sub-commands `create`, `update`, `describe`, `list`, and `delete`, to fully manage DomainMapping resources to use custom domains for Knative services.
```

```
# Create a domain mappings 'hello.example.com' for Knative service 'hello'
kn domain create hello.example.com --ref hello

# Update a domain mappings 'hello.example.com' for Knative service 'hello'
kn domain create hello.example.com --refFlags hello

# List all domain mappings
kn domain list

# Delete domain mappings 'hello.example.com'
kn domain delete hello.example.com
```
See `kn domain help` for more information.

> 

Reviewer: @csantanapr can you please open a docs issue for `kn domain` to get it properly documented in the site so it's not only explained here?

@rhuss can ya'll in the CLI group please come up with a process to make sure changes to the CLI like this are being communicated properly to docs? Otherwise they just don't get documented. CC @omerbensaadon

Reviewer 6: Sorry for that. I will update the PR template to raise awareness, so that doc relevant PR should be properly communicated.

What would be the best way to communicate ? A ping within the PR or creating an issue in the docs repo ?

Reviewer: @rhuss creating an issue in the docs repo would be perfect, thank you so much 🙂

Author: issues https://github.com/knative/docs/issues/3496 opened @abrennan89

#### Client API signature change

> A `context.Context` object is added as the first argument to every Client API method to align with Kubernetes' client API signatures. This change does not affect any client's CLI usage, only if the client-specific Golang API is used for communicating with the Knative backend has changed. The migration is straight forward, either pass an already existing context down to the call or leverage one of the available standard contexts (like `context.TODO()`).

Reviewer: Be consistent with "Client API", it's "client API" earlier in the doc, please choose one and use throughout.

Author: Just to make you aware I didn't write this content. 
Is a copy and paste verbatim from the release notes, this blog post is just trying to put all existing release notes in a single place. The only change I do is some minor adjustments. 

https://github.com/knative/client/releases/tag/v0.22.0
![image](https://user-images.githubusercontent.com/1094878/115811618-06e95200-a3be-11eb-98b1-0dea36a797aa.png)

I think is @rhuss who writes the release notes for client

Author: I'm glad that tech writers are reviewing this notes, I will be happy to make any suggested changes, and then I can open an issue to the respective repos for someone with write access to release notes to update them there.

> A `context.Context` object is added as the first argument to every Client API method to align with Kubernetes' client API signatures. This change does not affect any client's CLI usage, only if the client-specific Golang API is used for communicating with the Knative backend has changed. The migration is straight forward, either pass an already existing context down to the call or leverage one of the available standard contexts (like `context.TODO()`).

Reviewer: ```suggestion
A `context.Context` object was added as the first argument to every Client API method to align with the Kubernetes client API signatures. This change does not affect client CLI usage unless the client-specific Golang API used to communicate with the Knative backend has changed.

To migrate to the updated API signature, you can pass an already existing context to the call, or use one of the available standard contexts, such as `context.TODO()`.
```

This change does _not_ affect any CLI usage of the client.

### CLI Plugins

#### 💫 New Features & Changes

The plugins that are released aligned with 0.22 are:

- [kn-plugin-admin](https://github.com/knative-sandbox/kn-plugin-admin) for managing Knative installations that are running on Kubernetes | [download](https://github.com/knative-sandbox/kn-plugin-admin/releases/tag/v0.22.0)

> - [kn-plugin-source-kafka](https://github.com/knative-sandbox/kn-plugin-source-kafka) for managing a Kafka Source that has been installed via [eventing-kafka](https://github.com/knative-sandbox/eventing-kafka) on the backend | [download](https://github.com/knative-sandbox/kn-plugin-source-kafka/releases/tag/v0.22.0)

Reviewer: The new plugins are mentioned more generally in a section earlier in the doc; let's group this information maybe? It seems confusing to have it in different places.

Author: They are mentioned in the first section `Highlights` in one sentence.
```
- There are two new CLI plugins align with 0.22 release, [kn-plugin-admin](https://github.com/knative-sandbox/kn-plugin-admin) and [kn-plugin-source-kafka](https://github.com/knative-sandbox/kn-plugin-source-kafka)
```

This section is the area with the content details about the plugins. 

Let me know if you want me to remove the sentence from the first section `Highlights`

> ### Minor CLI updates
> 
> - Fix `kn export` uses the _Export_ format as the default
> - Added `S` column to specify built-in sources in `kn source list-types`
> - Added support for namespaces for all commands that takes a `--sink` option

Reviewer: ```suggestion
### Minor CLI updates

- `kn export` now uses the _Export_ format as the default.
- Added `S` column to specify built-in sources in `kn source list-types`.
- Added support for namespaces for all commands that takes a `--sink` option.
```

### Operator v0.22

#### 🐞 Bug Fixes

> - Delete the installed ingress resources only [#548](https://github.com/knative/operator/pull/548)
> - Allow the update of ingress resources with spec.additionalManifests [#531](https://github.com/knative/operator/pull/531)
> - Refactor the cache mechanism [#532](https://github.com/knative/operator/pull/532)
> - Filter the redundant resources in the target manifest [#509](https://github.com/knative/operator/pull/509)

Reviewer: ```suggestion
- Delete installed ingress resources only. [#548](https://github.com/knative/operator/pull/548)
- Allow the update of ingress resources with `spec.additionalManifests`. [#531](https://github.com/knative/operator/pull/531)
- Refactored the cache mechanism. [#532](https://github.com/knative/operator/pull/532)
- Filter redundant resources in the target manifest. [#509](https://github.com/knative/operator/pull/509)
```
